src/main/kotlin/no/nav/syfo/identendring/IdentendringService.kt

```diff
@@ -23,12 +24,20 @@ import java.time.LocalDate
 
 class IdentendringService(
     private val database: DatabaseInterface,
-    private val sendtSykmeldingKafkaProducer: SendtSykmeldingKafkaProducer
+    private val sendtSykmeldingKafkaProducer: SendtSykmeldingKafkaProducer,
+    private val pdlService: PdlPersonService
 ) {
-    fun oppdaterIdent(identListe: List<Ident>): Int {
+    suspend fun oppdaterIdent(identListe: List<Ident>): Int {
         if (harEndretFnr(identListe)) {
             val nyttFnr = identListe.find { it.type == IdentType.FOLKEREGISTERIDENT && it.gjeldende }?.idnummer
                 ?: throw IllegalStateException("Mangler gyldig fnr!")
+
+            val pdlPerson = pdlService.getPdlPerson(nyttFnr)
+
+            if (pdlPerson.fnr != nyttFnr) {
+                throw IllegalStateException("Nytt FNR er ulikt aktivt FNR i PDL API")
```

Comment:
Hva skjer når vi kaster feil her? Vil appen restarte, eller bare consumeren? Logges det errors, som igjen vil trigge alarmer? For vi ønsker vel egentlig ikke alarmer på dette, vi vil bare at den skal vente litt og prøve igjen :)

Jeg tror heller ikke sjekken vil virke. Hvis du slår opp nytt fnr og PDL ikke er oppdatert så vil kallet feile, så du kommer ikke til denne sjekken :)

Reply:
Godt poeng! Takk!

Reply:
Vi restarter bare consumeren når feil kastes der inne :)

Reply:
Så bra! Men er det noe errorlogging noe sted som vil trigge alarmer..?

Reply:
Ja, jeg skal prøve å gjøre det litt smartere. :)